readthedocs/core/history.py

```diff
@@ -0,0 +1,83 @@
+from functools import partial
+
+from django import forms
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from simple_history.admin import SimpleHistoryAdmin
+from simple_history.models import HistoricalRecords
+from simple_history.utils import update_change_reason
+
+
+class ExtraFieldsHistoricalModel(models.Model):
+
+    """
+    Abstract model to allow history models track extra data.
+
+    Extra data includes:
+
+    - Users after they have been deleted.
+    """
+
+    extra_history_user_id = models.IntegerField(
+        _('ID'),
+        blank=True,
+        null=True,
+        db_index=True,
+    )
+    extra_history_user_username = models.CharField(
+        _('username'),
+        max_length=150,
+        null=True,
+        db_index=True,
+    )
```

Comment:
We can track the ip and user agent here too, we have access to the request from the signal